checker/check_result.go

```diff
@@ -25,12 +26,44 @@ const MaxResultConfidence = 10
 // ErrorDemoninatorZero indicates the denominator for a proportional result is 0.
 var ErrorDemoninatorZero = errors.New("internal error: denominator is 0")
 
+// Types of details.
+const (
+	DetailFail = 0
+	DetailPass = 1
+	DetailInfo = 2
+	DetailWarn = 3
+)
+
+// CheckDetail contains information for each detail.
+//nolint:govet
+type CheckDetail struct {
+	Type int    // Any of DetailFail, DetailPass, DetailInfo.
+	Code string // A 4 digit string identifying the code, e.g. for remediation.
+	Desc string // A short string representation of the information.
+}
+
+func (cd *CheckDetail) Validate() {
+	if cd.Type < DetailFail ||
+		cd.Type > DetailWarn {
+		panic(fmt.Sprintf("invalid CheckDetail type: %v", cd.Type))
+	}
+}
+
+// Types of results.
+const (
+	ResultPass     = 0
+	ResultFail     = 1
+	ResultDontKnow = 2
+)
+
 type CheckResult struct {
```

Comment:
FYI, updating `CheckResult` schema will require a significant amount of work. At minimum, doing any changes here will require:

1. Updating cron job code which outputs JSON.
2. Updating BQ schema.
3. Following up with existing clients of BQ.

Reply:
Spoke too soon. Just add `json:"-"` to the new fields to avoid doing any of this.

Reply:
Thanks